### PR TITLE
Only test disabled-items-warnings if application is in editable state.

### PR DESCRIPTION
**Pull request by foxlynx**
New, updated story with more description: https://github.com/CSCfi/rems/issues/1002

### CODE REVIEW

File: src/cljs/rems/application.cljs [hunk 1]
```diff
@@ -765,7 +765,7 @@
                        (map dynamic-event->event (:dynamic-events app)))
         applicant-attributes (:applicant-attributes application)
         messages (remove nil?
-                         [(disabled-items-warning (:catalogue-items application)) ; NB: eval this here so we get nil or a warning
+                         [(when (editable? application)(disabled-items-warning (:catalogue-items application))) ; NB: eval this here so we get nil or a warning
```
Comment (opqdonut) [inline src/cljs/rems/application.cljs]: I'd prefer a line break at `)(`, but at least add a space :)

Comment (opqdonut) [reply src/cljs/rems/application.cljs]: should the condition actually be "when application is a draft"?

**Comment by Macroz** [comment]
Why is this a solution to the problem found in migration tests?

File: src/cljs/rems/application.cljs (hunk 1 above)
Comment (foxlynx) [reply src/cljs/rems/application.cljs]: I thought one can somehow modify returned application still, or does it create a new draft from the returned?

Comment (opqdonut) [reply src/cljs/rems/application.cljs]: oh right, I guess editable is the right condition then

Comment (foxlynx) [reply src/cljs/rems/application.cljs]: We are gonna dig a bit deeper tomorrow irl how this feature should be overall.

File: src/cljs/rems/application.cljs [hunk 2]
```diff
@@ -773,7 +773,8 @@
                        (map dynamic-event->event (:dynamic-events app)))
         applicant-attributes (:applicant-attributes application)
         messages (remove nil?
-                         [(disabled-items-warning (:catalogue-items application)) ; NB: eval this here so we get nil or a warning
+                         [(when (editable? application)
```
Comment (Macroz) [inline src/cljs/rems/application.cljs]: This check should be moved inside `disabled-items-warning`. However, there are also other needs as discussed so let's specify this task a bit better together.

**Comment by Macroz** [comment]
Closes #1002

**Pull request merged**